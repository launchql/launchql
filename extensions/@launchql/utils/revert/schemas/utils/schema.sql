-- Revert schemas/utils/schema from pg

BEGIN;

DROP SCHEMA utils;

COMMIT;
