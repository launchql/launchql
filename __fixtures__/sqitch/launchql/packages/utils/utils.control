# utils extension
comment = 'skitch project'
default_version = '0.0.1'
module_pathname = '$libdir/utils'
requires = 'plpgsql,uuid-ossp,totp,pg-verify'
relocatable = false
superuser = false
