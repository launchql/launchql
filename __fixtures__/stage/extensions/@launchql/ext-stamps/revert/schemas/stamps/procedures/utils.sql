-- Revert schemas/stamps/procedures/utils from pg

BEGIN;

DROP FUNCTION stamps.utils;

COMMIT;
