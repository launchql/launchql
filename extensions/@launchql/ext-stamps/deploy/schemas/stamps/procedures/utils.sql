-- Deploy schemas/stamps/procedures/utils to pg

-- requires: schemas/stamps/schema

BEGIN;

CREATE FUNCTION stamps.peoplestamps()
RETURNS TRIGGER AS $$
BEGIN
    IF TG_OP = 'INSERT' THEN
      NEW.created_by = jwt_public.current_user_id();
      NEW.updated_by = jwt_public.current_user_id();
    ELSIF TG_OP = 'UPDATE' THEN
      NEW.created_by = OLD.created_by;
      NEW.updated_by = jwt_public.current_user_id();
    END IF;
    RETURN NEW;
END;
$$ LANGUAGE 'plpgsql';

CREATE FUNCTION stamps.timestamps()
RETURNS TRIGGER AS $$
BEGIN
    IF TG_OP = 'INSERT' THEN
      NEW.created_at = NOW();
      NEW.updated_at = NOW();
    ELSIF TG_OP = 'UPDATE' THEN
      NEW.created_at = OLD.created_at;
      NEW.updated_at = NOW();
    END IF;
    RETURN NEW;
END;
$$ LANGUAGE 'plpgsql';

COMMIT;
