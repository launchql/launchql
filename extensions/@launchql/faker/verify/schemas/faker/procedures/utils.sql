-- Verify schemas/faker/procedures/utils  on pg

BEGIN;

SELECT verify_function ('faker.utils');

ROLLBACK;
