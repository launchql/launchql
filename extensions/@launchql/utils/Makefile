EXTENSION = launchql-utils
DATA = sql/launchql-utils--0.4.6.sql

PG_CONFIG = pg_config
PGXS := $(shell $(PG_CONFIG) --pgxs)
include $(PGXS)
  