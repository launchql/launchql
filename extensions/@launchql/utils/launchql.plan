%syntax-version=1.0.0
%project=launchql-utils
%uri=launchql-utils
  
schemas/utils/schema 2017-08-11T08:11:51Z skitch <skitch@5b0c196eeb62> # add schemas/utils/schema
schemas/utils/procedures/mask_pad [schemas/utils/schema] 2017-08-11T08:11:51Z skitch <skitch@5b0c196eeb62> # add schemas/utils/procedures/mask_pad
schemas/utils/procedures/throw [schemas/utils/schema] 2017-08-11T08:11:51Z skitch <skitch@5b0c196eeb62> # add schemas/utils/procedures/throw
schemas/utils/procedures/ensure_singleton [schemas/utils/schema] 2021-04-25T11:50:15Z Dan Lynch <dlynch@Dans-MBP-3> # add schemas/utils/procedures/ensure_singleton
