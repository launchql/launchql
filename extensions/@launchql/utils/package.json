{
  "_from": "@launchql/utils@^0.4.6",
  "_id": "@launchql/utils@0.4.6",
  "_inBundle": false,
  "_integrity": "sha512-1ZB6FYCWUCr0casWo7m8Xr74IGpwVSlaTXGnVhslXoLAaU/nwIr39khzRRkHKIrjJNG4VoWfMTymZLfDAQDj1Q==",
  "_location": "/@launchql/utils",
  "_phantomChildren": {},
  "_requested": {
    "type": "range",
    "registry": true,
    "raw": "@launchql/utils@^0.4.6",
    "name": "@launchql/utils",
    "escapedName": "@launchql%2futils",
    "scope": "@launchql",
    "rawSpec": "^0.4.6",
    "saveSpec": null,
    "fetchSpec": "^0.4.6"
  },
  "_requiredBy": [
    "/@pyramation/launchql-meta-schema"
  ],
  "_resolved": "https://registry.npmjs.org/@launchql/utils/-/utils-0.4.6.tgz",
  "_shasum": "4d30448b4bc2efa4891add0b9d393c66ad181333",
  "_spec": "@launchql/utils@^0.4.6",
  "_where": "/private/var/folders/kn/177lldss4n9fwn0gh5q5wfn00000gn/T/ny4sag54vjfki5oa3m947/extensions/node_modules/@pyramation/launchql-meta-schema",
  "author": {
    "name": "Dan Lynch",
    "email": "pyramation@gmail.com"
  },
  "bundleDependencies": false,
  "deprecated": false,
  "description": "utils",
  "devDependencies": {
    "@babel/cli": "7.10.4",
    "@babel/core": "7.10.4",
    "@babel/plugin-proposal-class-properties": "7.10.4",
    "@babel/plugin-proposal-export-default-from": "7.10.4",
    "@babel/plugin-proposal-object-rest-spread": "7.10.4",
    "@babel/plugin-transform-runtime": "7.10.4",
    "@babel/preset-env": "7.10.4",
    "@babel/runtime": "^7.4.2",
    "@launchql/cli": "1.3.5",
    "@launchql/db-testing": "latest",
    "babel-eslint": "10.1.0",
    "babel-jest": "26.1.0",
    "eslint": "^7.3.1",
    "eslint-config-prettier": "^6.10.0",
    "eslint-plugin-prettier": "^3.1.2",
    "jest": "26.1.0",
    "prettier": "2.0.5",
    "regenerator-runtime": "^0.13.2"
  },
  "gitHead": "7e686ca8bfee944da876b6cfd2c32e71da244537",
  "name": "@launchql/utils",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "prepare": "lql package",
    "test": "FAST_TEST=1 launchql-templatedb && jest",
    "test:watch": "FAST_TEST=1 jest --watch"
  },
  "version": "0.4.6"
}
