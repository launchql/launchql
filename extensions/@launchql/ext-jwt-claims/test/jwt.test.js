import { getConnections } from './utils';

let db, teardown;
const apis = {};
const jwt = {
  user_id: 'b9d22af1-62c7-43a5-b8c4-50630bbd4962',
  database_id: '44744c94-93cf-425a-b524-ce6f1466e327'
};
beforeAll(async () => {
  ({ db, teardown } = await getConnections());
  apis.public = db.helper('jwt_public');
  apis.private = db.helper('jwt_private');
});

afterAll(async () => {
  try {
    //try catch here allows us to see the sql parsing issues!
    await teardown();
  } catch (e) {
    // noop
  }
});

beforeEach(async () => {
  await db.beforeEach();
});

afterEach(async () => {
  await db.afterEach();
});

it('get values', async () => {
  db.setContext({
    'jwt.claims.user_agent':
      'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36',
    'jwt.claims.ip_address': '127.0.0.1',
    'jwt.claims.database_id': jwt.database_id,
    'jwt.claims.user_id': jwt.user_id
  });
  const user_agent = await apis.public.call('current_user_agent');
  const ip_address = await apis.public.call('current_ip_address');
  const database_id = await apis.private.call('current_database_id');
  const user_id = await apis.public.call('current_user_id');
  expect({ user_agent }).toMatchSnapshot();
  expect({ ip_address }).toMatchSnapshot();
  expect({ database_id }).toMatchSnapshot();
  expect({ user_id }).toMatchSnapshot();
});
