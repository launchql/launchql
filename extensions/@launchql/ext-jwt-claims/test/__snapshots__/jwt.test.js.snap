// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get values 1`] = `
Object {
  "user_agent": Array [
    Object {
      "current_user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36",
    },
  ],
}
`;

exports[`get values 2`] = `
Object {
  "ip_address": Array [
    Object {
      "current_ip_address": "127.0.0.1",
    },
  ],
}
`;

exports[`get values 3`] = `
Object {
  "database_id": Array [
    Object {
      "current_database_id": "44744c94-93cf-425a-b524-ce6f1466e327",
    },
  ],
}
`;

exports[`get values 4`] = `
Object {
  "group_ids": Array [
    Object {
      "current_group_ids": Array [
        "f12c75c2-47d5-43fd-9223-d42d08f51942",
        "d96d32b4-e819-4cb1-8a27-e27e763e0d7f",
        "c8a27b31-1d40-4f40-9cb0-e96a44e68072",
      ],
    },
  ],
}
`;

exports[`get values 5`] = `
Object {
  "user_id": Array [
    Object {
      "current_user_id": "b9d22af1-62c7-43a5-b8c4-50630bbd4962",
    },
  ],
}
`;
